### ISSUE TITLE
NPC Object Type

### ISSUE THREAD

**Issue by Nix-Starlight** [CoagulateSL/GPHUD#106]
Put a script into an object, it becomes an NPC automatically. Popups to the script owner to select the stats for the NPC. It's given the same options as a player. Anyone with the right permission group or group can change the stats. If a player can attack another player for example, then the NPC will show up on their radar the same as any other player with the given inputted name.

**Comment by iprice** [comment]
NPC is already an object type.
The first half is how you set up an NPC object type already (be the NPC, configure it, make it an NPC).

Permissions already exist on the global scale ; you want an option to allow just limited users to edit limited NPCs?

Otherwise the "shown on the avatar radar" is problematic due to the asynchronous nature of things, that is:

Currently, to get a list of avatars, the HUD calls a "Sensor" that, some time later, returns a list of all avatars in range.

In order to include objects,  one has to send out the sensor, AND send out a local chat message for local objects to reply to
One then has to wait an "undetermined" ammount of time for replies ; how fast do scripts respond on your sim, and how fast do scripts respond on the worst sim that uses my product?  Now ever single player targetting popup has to wait this long for object to reply, wether they exist or not.

OR.  sensor can detect scripted objects, but will detect ALL scripted objects including e.g. poseballs.  It has a limit of 16 responses so if number of avatars + objects > 16 then maybe avatars would be dropped.

In short, doesn't seem a "good" solution to this.  Other than having all the NPCs constantly register with the HUD, but I dont think it has the memory to store enough NPCs for some sims.
